### PR TITLE
Initial

### PR DESCRIPTION
**Pull request by dlongley**
Note, no tests here.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+  env: {
```
Comment (davidlehn) [inline .eslintrc.js:3]:
```suggestion
  root: true,
  env: {
```

**Review by gannan08 — COMMENTED** [review]
Need an issue to create a test suite. How are errors handled in this paradigm?

**Comment by dlongley** [comment]
Test suite issue: #2.

> How are errors handled in this paradigm?

Emitters try/catch and do whatever they want in their application as in the example.

**Review by gannan08 — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -1,2 +1,114 @@
...
+  methods: {
+    foo(event) {
+      event.respondWith(this.doSomethingAsync());
+    },
+    async doSomethingAsync() {
+      // do something async and return something for the emitter
+    }
+  }
+}
+```
```
Comment (gannan08) [inline README.md:114]: A quick warning here, one may read too much into `foo`. It is possible that one may think that they need to name everything after the name of the event... even the key, in the object, that gets emitted. Of course, that is not the case.

We will need to make a more realistic example for developers. This updated example can be done in another PR.

**Pull request merged**